topbar.js

```diff
@@ -622,8 +623,14 @@ function setWorkspaceName (name) {
 }
 
 function setMonitor(monitor) {
-    if (!prefs.topbar_follow_focus)
-        monitor = Main.layoutManager.primaryMonitor;
```

Comment:
This needs to happen to. `setMonitor` is called when changing workspaces, and without it `panelMonitor` will be incorrect.